src/Connection.php

```diff
@@ -325,4 +332,37 @@ public function getOption($name)
     {
         return isset($this->options[$name]) ? $this->options[$name] : null;
     }
+
+    /**
+     * Implements ConnectionInterface::setData().
+     *
+     * @param string $name Data key
+     * @param mixed $value Data value
+     * @return $this
+     */
+    public function setData($name, $value)
+    {
+        $this->data[$name] = $value;
+
+        return $this;
+    }
+
+    /**
+     * Implements ConnectionInterface::getData().
+     *
+     * @param string $name Data key
+     * @return mixed
+     */
+    public function getData($name)
+    {
+        return isset($this->data[$name]) ? $this->data[$name] : null;
+    }
+
+    /**
+     * Implements ConnectionInterface::clearData().
+     */
+    public function clearData()
```

Comment:
Seems like `clearData()` should implement a fluent interface as `setData()` does.